### DISCUSSION TITLE
JobRunr OSS (free) and CRON jobs - Updated!

### DISCUSSION THREAD

**Discussion by rdehuyss** [jobrunr/jobrunr#465] [Polls]
Hi all,

Recently, I'm getting a lot of issues from users that have a lot of CRON jobs (see #437, #462, #460). 

**A little bit of background**: the projects I worked on (for which I created JobRunr) were mostly integration projects where we need to check a file share every 5 minutes and if there are files available, we then do create millions of jobs to import all the data. I initially thought that 20 recurring jobs in a project was a lot 🤣 . That's why the recurring jobs page does not have paging. 

> Now I see users are doing 1700 🚀 CRON jobs or more.

Although JobRunr starts to have some happy Pro customers, the amount of unpaid time I'm spending on this project, is tremendous. However, I still need to make a living and support my family financially. Combining JobRunr with another full-time job is starting to become a real burden.  I'm concerned that at the current pace, a burn-out may be looming in the future and I prefer to keep this project going on for a long time.

After quite some thinking on this, I found a way to support so many CRON jobs and keep performance top notch. But, I'm thinking of limiting the amount of CRON jobs per pricing plan.

- JobRunr OSS / free: max 100 CRON jobs
- JobRunr Startup Plan: max 250 CRON jobs
- JobRunr SME Plan: max 500 CRON jobs
- JobRunr Enterprise Plan: unlimited CRON jobs

But as I know that I'm **_taking away_** features from JobRunr, I'm open to discuss this. If you have a better idea on how to deal with my actual problem (spending a looooooot of unpaid time after my day to day job that impacts my personal & family life), feel free to comment. Or even better, feel free to comment either way. All your input is appreciated.

Thanks,
Ronald

**Comment by rdehuyss** [comment]
I've updated the poll to change the amount of free CRON jobs from 20 to 100. That's why the votes are gone :-s.

For those people that will be building their own JobRunr: how many recurring jobs are you currently running?

**Comment by dmurphy-ambra** [comment]
I certainly sympathize with your situation @rdehuyss and your desire to find a sustainable path forward.  From my standpoint, when the broader open source community is finding issues due to scalability, they are actually doing you, and JobRunr Pro licensees a big favor.  The big benefit of open source is all of us as a worldwide community finding issues and making things better.   If you cap the number of jobs in the OS version then the people that will find these issues are Pro customers, likely in Production.... ouch :)

On the other hand, I see no issue with you saying that JobRunr works best with up to X recurring jobs, and after that you may run into problems.  This way you can acknowledge the issue, but take your time solving it.  If a Pro customer is hitting the issue that's different and maybe you prioritize it.

**Comment by Mumeii** [comment]
Hi @rdehuyss !

A quick question regarding this poll : did it already helped you deciding how you want to change your JobRunR Pro business model ?

I don't see any references regarding a cron job limitation on the [JobRunR Pro pricing page](https://www.jobrunr.io/en/pricing/), so I'm wondering if you've changed your mind about it, or if we should forecast a change soon regarding this new limitation ?

During my current POC I'm performing tests relevant for a target production environment which will have around 10 to 15 batches, most of which should split their work amongst our ~1700 stores. We're aiming to split our jobs this way in order to benefit from the zoneId feature for each of them. 

So we'll quickly reach a ~20,000 or more recurring job target, and it could keep increasing if we keep opening stores.

Regarding this POC, I'm not in charge of the final decision, especially regarding the financial part, but I have to make a rough forecast of how much it could cost if we go for JobRunR.

So having your feedback about it would be great 😃

**Reply by rdehuyss** [reply]
Hi @Mumeii - I emailed you!

Anyway, this is fixed in the Pro version and as I see how the voting went, I will probably keep the amounts mentioned here above.
